components/wifi/esp8266-driver/ESP8266Interface.cpp

```diff
@@ -224,13 +226,15 @@ void ESP8266Interface::_connect_async()
     nsapi_error_t status = _init();
     if (status != NSAPI_ERROR_OK) {
         _connect_retval = status;
+        _esp.uart_enable_input(false);
```

Comment:
Uart input enabling/disabling depends on ` _software_conn_stat ` value, so it would be better to enable or disable input once when it changes. `_init` should rather rely on the driver to have uart input enabled (depending on the `conn_stat`), than try and influence it on its own. 
Also, in this way we will avoid repetition of the same condition statement before each return.  
@VeijoPesonen, does this make sense?

Reply:
> Uart input enabling/disabling depends on _software_conn_stat value, so it would be better to enable or disable input once when it changes.

Ok, point taken :)

> Also, in this way we will avoid repetition of the same condition statement before each return.

Not needed based on your response.